torch/_dynamo/variables/inline_helper.py

```diff
@@ -0,0 +1,167 @@
+import warnings
+
+import torch.fx
+import torch.fx.traceback as fx_traceback
+import torch.utils._pytree as pytree
+from torch.fx import Interpreter
+from torch.nn.utils import stateless
+from .constant import ConstantVariable
+
+
+def dummy_user_function_to_inline_gm(gm, args):
+    return gm(*args)
+
+
+def dummy_user_function_to_inline_wrapped_gm(wrapped_gm, args, kwargs):
+    return wrapped_gm(args, kwargs)
+
+
+def dummy_accumulate_grad_(t1, t2):
+    if t1.grad is None:
+        t1.grad = t2
+    else:
+        t1.grad += t2
+
+
+# copy from torch/_functorch/_aot_autograd/traced_function_transforms.py
+# and remove the restriction that output has to be a tuple.
```

Comment:
cc @bdhirsh can we upstream this

Reply:
yeah (Jack if you just upstream but keep the assert in AOTAutograd directly, feel free to change in this PR)

Reply:
Yea Let me follow up.